### PR TITLE
Add flag and rebuild test package docs

### LINKED ISSUE #1456: dartdoc integration test needs to have a mode that doesn't fail on trivial doc changes
To fix the test, I suggest adding a dartdoc command line option to drop inherited text from the SDK in generated docs and replace it with a stub.  This way we can still verify that dartdoc is generating the right structure and is choosing correctly when to import docs from the SDK without depending on the exact contents of those docs themselves.

I believe this is the main blocker on running our tests as part of the main suite in the SDK.

@devoncarew

### PR DESCRIPTION
**Pull request by jcollins-g**
Fixes #1456 and should enable dartdoc tests to work reliably in the SDK.

As discussed before, add a flag to hide the imported SDK docs in the integration tests.

### CODE REVIEW

File: bin/dartdoc.dart [hunk 1]
```diff
@@ -266,6 +289,14 @@ ArgParser _createArgsParser() {
           "Generates `index.json` with indentation and newlines. The file is larger, but it's also easier to diff.",
       negatable: false,
       defaultsTo: false);
+  parser.addFlag(
+    'hide-sdk-text',
+    help:
```
Comment (devoncarew) [inline bin/dartdoc.dart]: add `hide: true`?

File: lib/src/model.dart [hunk 2]
```diff
@@ -1956,17 +1956,24 @@ abstract class ModelElement implements Comparable, Nameable, Documentable {
...
+      _rawDocs = _injectMacros(_rawDocs);
+    }
     return _rawDocs;
   }
 
   /// Returns the docs, stripped of their leading comments syntax.
   @override
-  String get documentation => documentationFrom._documentationLocal;
+  String get documentation {
+    String x = documentationFrom._documentationLocal;
```
Comment (devoncarew) [inline lib/src/model.dart]: use the function shorthand?

File: lib/src/config.dart [hunk 3]
```diff
@@ -38,10 +40,14 @@ void setConfig(
     bool includeSource: true,
     String sdkVersion,
     bool autoIncludeDependencies: false,
-    List<String> categoryOrder}) {
+    List<String> categoryOrder,
+    List<String> dropTextFrom}) {
   if (categoryOrder == null) {
     categoryOrder = new UnmodifiableListView<String>([]);
   }
+  if (dropTextFrom == null) {
```
Comment (devoncarew) [inline lib/src/config.dart]: `??=`

**Comment by devoncarew** [comment]
lgtm

File: bin/dartdoc.dart (hunk 1 above)
Comment (jcollins-g) [inline bin/dartdoc.dart]: whoops.  Fixed.

File: lib/src/config.dart (hunk 3 above)
Comment (jcollins-g) [inline lib/src/config.dart]: done

File: lib/src/model.dart (hunk 2 above)
Comment (jcollins-g) [inline lib/src/model.dart]: Oops.  I keep forgetting to undo my manual conditional breakpoints completely.

**Pull request merged**